js/clinicalTimeline.js

```diff
@@ -884,10 +899,29 @@ window.clinicalTimeline = (function(){
               tickValues.push(i * maxTime.daysPerYear);
           }
       } else if (zoomLevel === "months") {
-          tickValues.push(parseInt(beginning));
-          for (i=minTime.m + minTime.y * 12 + 1; i < maxTime.m + maxTime.y * 12 - 1; i++) {
+          
+          if (minTime.d !== 0) {
+            beginning -= minTime.d < 0 ? 30 : -30;
+            minTime = daysToTimeObject(parseInt(beginning));
+            timePeriod = daysToTimeObject(difference(parseInt(beginning), parseInt(ending)));
+          }
+
+          if (maxTime.d !== 0) {
+            ending += maxTime.d > 0 ? 30 : -30;
+            maxTime = daysToTimeObject(parseInt(ending));
+            timePeriod = daysToTimeObject(difference(parseInt(beginning), parseInt(ending)));
+          }
+
+          for (i=minTime.m + minTime.y * 12; i <= maxTime.m + maxTime.y * 12 ; i++) {
               tickValues.push(i * maxTime.daysPerMonth + parseInt(i/12) * 5);
           }
+
+          beginning = tickValues[0];
+          ending = tickValues[tickValues.length-1];
+          minTime = daysToTimeObject(parseInt(beginning));
+          maxTime = daysToTimeObject(parseInt(ending));
+          timePeriod = daysToTimeObject(difference(parseInt(beginning), parseInt(ending)));
```

Comment:
Not entirely sure why you would need to keep track of all those variables above. Are you still using those after? It's also better to not change the variables passed to the function i.e. beginning and ending or clarity's sake. Furthermore it seems that timePeriod is never used so we should prolly get rid of that variable completely.